In [4]:
import pandas as pd

titanic_df = pd.read_csv(
    r"C:\Users\shahh\OneDrive\Desktop\Data Science And Machine Learning\Skill_Sikshya\Day22\titanic_data.csv",
    delimiter = ",",
    header = 0
    )

titanic_df


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S
...,...,...,...,...,...,...,...,...,...,...,...,...
886,887,0,2,"Montvila, Rev. Juozas",male,27.0,0,0,211536,13.0000,NaN,S
887,888,1,1,"Graham, Miss. Margaret Edith",female,19.0,0,0,112053,30.0000,B42,S
888,889,0,3,"Johnston, Miss. Catherine Helen ""Carrie""",female,NaN,1,2,W./C. 6607,23.4500,NaN,S
889,890,1,1,"Behr, Mr. Karl Howell",male,26.0,0,0,111369,30.0000,C148,C


Basic Information


In [5]:
titanic_df.shape

(891, 12)

In [6]:
titanic_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    str    
 4   Sex          891 non-null    str    
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    str    
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    str    
 11  Embarked     889 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 83.7 KB


The Titanic dataset contains 891 passenger records and 12 columns, including age, gender, passenger class, fare, and survival status. It contains some missing values, particularly in Age, Cabin, and Embarked. The dataset is widely used for practicing Pandas, data analysis, and machine learning.

Specific Information


In [15]:
titanic_df.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


In [7]:
# Stat info : mean Q1 median Q3 max min mode std corr
titanic_df[["Age", "Fare"]].mean()

Age     29.699118
Fare    32.204208
dtype: float64

In [8]:
titanic_df[["Age", "Fare"]].median()

Age     28.0000
Fare    14.4542
dtype: float64

In [9]:
titanic_df[["Age", "Fare"]].std()

Age     14.526497
Fare    49.693429
dtype: float64

In [10]:
titanic_df[["Age", "Fare"]].mode()

,Age,Fare
0,24.0,8.05


In [11]:
titanic_df[["Age", "Fare"]].count()

Age     714
Fare    891
dtype: int64

In [12]:
titanic_df[["Age", "Fare"]].var()

Age      211.019125
Fare    2469.436846
dtype: float64

In [13]:
titanic_df[["Age", "Fare"]].min()

Age     0.42
Fare    0.00
dtype: float64

In [14]:
titanic_df[["Age", "Fare"]].max()

Age      80.0000
Fare    512.3292
dtype: float64

In [20]:
titanic_df[["Age", "Fare"]].describe()

,Age,Fare
count,714.000000,891.000000
mean,29.699118,32.204208
std,14.526497,49.693429
min,0.420000,0.000000
25%,20.125000,7.910400
50%,28.000000,14.454200
75%,38.000000,31.000000
max,80.000000,512.329200


Titanic Dataset – Age & Fare Analysis

• The dataset contains 891 passengers.
• Age has 714 valid values and 177 missing values.
• The average passenger age is 29.7 years.
• The median passenger age is 28 years.
• Passenger ages range from 0.42 to 80 years.
• The average fare is 32.20.
• The median fare is 14.45.
• Fare values range from 0 to 512.33.
• The difference between mean and median fare shows that Fare is right-skewed.
• The maximum fare of 512.33 indicates the presence of outliers.
• 75% of passengers paid a fare of 31 or less.
• Age and Fare show considerable variation among passengers.
• Missing Age values should be handled before further analysis or machine learning.

In [22]:
# Counts
titanic_df.value_counts("Embarked")

Embarked
S    644
C    168
Q     77
Name: count, dtype: int64

In [27]:
titanic_df.value_counts("Embarked", normalize = True, ascending = False, dropna = True)

Embarked
S    0.724409
C    0.188976
Q    0.086614
Name: proportion, dtype: float64

In [30]:
titanic_df.value_counts("Cabin", normalize = False, dropna = True)

Cabin
G6             4
C23 C25 C27    4
B96 B98        4
F33            3
E101           3
              ..
E17            1
A24            1
C50            1
B42            1
C148           1
Name: count, Length: 147, dtype: int64

In [32]:
titanic_df.value_counts(["Sex", "Survived"], normalize = True)

Sex     Survived
male    0           0.525253
female  1           0.261504
male    1           0.122334
female  0           0.090909
Name: proportion, dtype: float64

In [33]:
titanic_df[titanic_df.Sex == "female"].value_counts("Survived")

Survived
1    233
0     81
Name: count, dtype: int64

In [36]:
# Embarked from s , Fare Distribution
titanic_df[titanic_df.Embarked == "S"]["Fare"].describe()

count    644.000000
mean      27.079812
std       35.887993
min        0.000000
25%        8.050000
50%       13.000000
75%       27.900000
max      263.000000
Name: Fare, dtype: float64

In [ ]:
# embarked from S , male , survied
s_male = titanic_df[(titanic_df["Embarked"] == "S") & (titanic_df["Sex"] == "male")].value_counts("Survived")
s_male




Survived
0    364
1     77
Name: count, dtype: int64